popmon/config.py

```diff
@@ -16,66 +16,140 @@
-    "(i.e. has a high cardinality), then the most frequent values are displayed and the remaining are grouped as 'Others'. "
-    "The maximum number of values to should is configurable (default: 20).",
-    "heatmap_column_normalized": "The column-normalized heatmap allows for comparing of time bins when the counts in each bin vary.",
-    "heatmap_row_normalized": "The row-normalized heatmaps allows for monitoring one value over time.",
-}
-
-config = {
-    "section_descriptions": section_descriptions,
-    "limited_stats": [
+from pydantic import BaseModel, BaseSettings
+
+# Global configuration for the joblib parallelization. Could be used to change the number of jobs, and/or change
+# the backend from default (loki) to 'multiprocessing' or 'threading'.
+# (see https://joblib.readthedocs.io/en/latest/generated/joblib.Parallel.html for details)
+parallel_args = {"n_jobs": 1}
+
+# Usage the `ing_matplotlib_theme`
+themed = True
+
+
+class ProfilesSection(BaseModel):
+    name = "Profiles"
+    description = """Basic statistics of the data (profiles) calculated for each time period (a period
+                       is represented by one bin). The yellow and red lines represent the corresponding
+                       traffic light bounds (default: 4 and 7 standard deviations with respect to the reference data)."""
+
+
+class AlertSection(BaseModel):
+    name = "Alerts"
+    description = "Alerts aggregated by all traffic lights for each feature."
+
+    descriptions = {
+        "n_green": "Total number of green traffic lights (observed for all statistics)",
+        "n_yellow": "Total number of  yellow traffic lights (observed for all statistics)",
+        "n_red": "Total number of red traffic lights (observed for all statistics)",
+    }
+
+
+class HistogramSectionModel(BaseModel):
+    name = "Histograms"
+    description = "Histograms of the last few time slots (default: 2)."
+
+    hist_names: List[str] = [
+        "heatmap",
+        "heatmap_column_normalized",
+        "heatmap_row_normalized",
+    ]
+    hist_names_formatted = {
+        "heatmap": "Heatmap",
+        "heatmap_column_normalized": "Column-Normalized Heatmap",
+        "heatmap_row_normalized": "Row-Normalized Heatmap",
+    }
+    descriptions = {
+        "heatmap": "The heatmap shows the frequency of each value over time. If a variable has a high number of distinct values"
+        "(i.e. has a high cardinality), then the most frequent values are displayed and the remaining are grouped as 'Others'. "
+        "The maximum number of values to should is configurable (default: 20).",
+        "heatmap_column_normalized": "The column-normalized heatmap allows for comparing of time bins when the counts in each bin vary.",
+        "heatmap_row_normalized": "The row-normalized heatmaps allows for monitoring one value over time.",
+    }
+
+    """
+    plot_hist_n: plot histograms for last 'n' periods. default is 2 (optional)
```

Comment:
Check pydantic syntax for documentation

Reply:
Not supported by default 

There are third-party implementations
https://github.com/danields761/pydantic-settings

For now users can find the doc(strings) in the config file